File: brique-94/src/sdk/web/molam-form.js
/**
 * Molam Form Core - Web SDK
 * Universal checkout widget for merchant integration
 *
 * Usage:
 * <script src="https://cdn.molam.com/molam-form.js"></script>
 * <molam-checkout publishable-key="pk_test_xxx" amount="1000" currency="USD"></molam-checkout>
 */

(function() {
  'use strict';

  const SDK_VERSION = '1.0.0';
  const API_BASE_URL = 'https://api.molam.com/form'; // Production
  // const API_BASE_URL = 'http://localhost:3000/form'; // Development

  /**
   * Molam SDK Class
   */
  class MolamSDK {
    constructor(publishableKey) {
      if (!publishableKey || !publishableKey.startsWith('pk_')) {
        throw new Error('Invalid publishable key. Must start with "pk_"');
      }
      this.publishableKey = publishableKey;
      this.environment = publishableKey.startsWith('pk_test_') ? 'test' : 'live';
    }

    /**
     * Create a payment intent
     */
    async createPaymentIntent(params) {
      const { amount, currency, metadata, customer_email, customer_name, description, return_url } = params;

      const response = await fetch(`${API_BASE_URL}/payment-intents`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${this.publishableKey}`
        },
        body: JSON.stringify({
          amount,
          currency,
          metadata,
          customer_email,
          customer_name,
          description,
          return_url
        })
      });

      if (!response.ok) {
        const error = await response.json();
        throw new Error(error.message || 'Failed to create payment intent');
      }

      const intent = await response.json();
      await this.logEvent('intent_created', { intent_reference: intent.intent_reference });
      return intent;
    }

    /**
     * Confirm a payment intent
     */
    async confirmPaymentIntent(intentId, paymentMethodToken) {
      const response = await fetch(`${API_BASE_URL}/payment-intents/${intentId}`, {
        method: 'PATCH',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${this.publishableKey}`
        },
        body: JSON.stringify({
          action: 'confirm',
          payment_method_token: paymentMethodToken
        })
      });

      if (!response.ok) {
        const error = await response.json();
        throw new Error(error.message || 'Failed to confirm payment intent');
      }

      const intent = await response.json();
      await this.logEvent('intent_confirmed', { intent_reference: intent.intent_reference });
      return intent;
    }

    /**
     * Log telemetry event
     */
    async logEvent(eventType, payload = {}) {
      try {
        await fetch(`${API_BASE_URL}/logs`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${this.publishableKey}`
          },
          body: JSON.stringify({
            event_type: eventType,
            sdk_version: SDK_VERSION,
            platform: 'web',
            payload: payload
          })
        });
      } catch (error) {
        // Silent fail for logging
        console.warn('Failed to log event:', error);
      }
    }

    /**
     * Tokenize payment method (PCI-compliant)
     * In production, this would send card data to a PCI-compliant tokenization service
     */
    async tokenizePaymentMethod(cardData) {
      // Mock tokenization - in production, use a real tokenization service
      const token = `pm_${this.environment}_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`;

      await this.logEvent('payment_method_tokenized', {
        token_prefix: token.substring(0, 10),
        card_last4: cardData.number.slice(-4)
      });

      return {
        token,
        card: {
          last4: cardData.number.slice(-4),
          brand: this.detectCardBrand(cardData.number),
          exp_month: cardData.expMonth,
          exp_year: cardData.expYear
        }
      };
    }

    /**
     * Detect card brand from number
     */
    detectCardBrand(number) {
      const patterns = {
        visa: /^4/,
        mastercard: /^5[1-5]/,
        amex: /^3[47]/,
        discover: /^6(?:011|5)/
      };

      for (const [brand, pattern] of Object.entries(patterns)) {
        if (pattern.test(number)) return brand;
      }
      return 'unknown';
    }
  }

  /**
   * Custom Element: <molam-checkout>
   */
  class MolamCheckoutElement extends HTMLElement {
    constructor() {
      super();
      this.attachShadow({ mode: 'open' });
      this.sdk = null;
      this.paymentIntent = null;
    }

    connectedCallback() {
      const publishableKey = this.getAttribute('publishable-key');
      const amount = parseFloat(this.getAttribute('amount') || '0');
      const currency = (this.getAttribute('currency') || 'USD').toUpperCase();
      const customerEmail = this.getAttribute('customer-email');
      const customerName = this.getAttribute('customer-name');
      const description = this.getAttribute('description');

      if (!publishableKey) {
        this.showError('Missing publishable-key attribute');
        return;
      }

      if (amount <= 0) {
        this.showError('Invalid amount. Must be greater than 0');
        return;
      }

      try {
        this.sdk = new MolamSDK(publishableKey);
        this.render();
        this.initializePayment(amount, currency, customerEmail, customerName, description);
      } catch (error) {
        this.showError(error.message);
      }
    }

    async initializePayment(amount, currency, customerEmail, customerName, description) {
      try {
        this.showLoading();
        this.paymentIntent = await this.sdk.createPaymentIntent({
          amount,
          currency,
          customer_email: customerEmail,
          customer_name: customerName,
          description: description || `Payment of ${amount} ${currency}`,
          return_url: window.location.href
        });
        this.showPaymentForm();
      } catch (error) {
        this.showError(`Failed to initialize payment: ${error.message}`);
      }
    }

    render() {
      const styles = `
        <style>
          :host {
            display: block;
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, sans-serif;
            max-width: 400px;
            margin: 0 auto;
          }

          .molam-container {
            background: white;
            border: 1px solid #e0e0e0;
            border-radius: 8px;
            padding: 24px;
            box-shadow: 0 2px 8px rgba(0,0,0,0.1);
          }

          .molam-header {
            text-align: center;
            margin-bottom: 24px;
          }

          .molam-amount {
            font-size: 32px;
            font-weight: bold;
            color: #1a1a1a;
          }

          .molam-description {
            font-size: 14px;
            color: #666;
            margin-top: 8px;
          }

          .molam-form-group {
            margin-bottom: 16px;
          }

          .molam-label {
            display: block;
            font-size: 14px;
            font-weight: 500;
            color: #333;
            margin-bottom: 6px;
          }

          .molam-input {
            width: 100%;
            padding: 12px;
            border: 1px solid #d0d0d0;
            border-radius: 4px;
            font-size: 14px;
            box-sizing: border-box;
            transition: border-color 0.2s;
          }

          .molam-input:focus {
            outline: none;
            border-color: #5469d4;
            box-shadow: 0 0 0 3px rgba(84, 105, 212, 0.1);
          }

          .molam-input.error {
            border-color: #e74c3c;
          }

          .molam-card-row {
            display: flex;
            gap: 12px;
          }

          .molam-card-row .molam-form-group {
            flex: 1;
          }

          .molam-button {
            width: 100%;
            padding: 14px;
            background: #5469d4;
            color: white;
            border: none;
            border-radius: 4px;
            font-size: 16px;
            font-weight: 600;
            cursor: pointer;
            transition: background-color 0.2s;
          }

          .molam-button:hover {
            background: #4159c3;
          }

          .molam-button:disabled {
            background: #cccccc;
            cursor: not-allowed;
          }

          .molam-error {
            background: #fff5f5;
            border: 1px solid #feb2b2;
            border-radius: 4px;
            padding: 12px;
            color: #e74c3c;
            font-size: 14px;
            margin-bottom: 16px;
          }

          .molam-success {
            background: #f0fdf4;
            border: 1px solid #86efac;
            border-radius: 4px;
            padding: 12px;
            color: #16a34a;
            font-size: 14px;
            text-align: center;
          }

          .molam-loading {
            text-align: center;
            padding: 40px;
            color: #666;
          }

          .molam-spinner {
            border: 3px solid #f3f3f3;
            border-top: 3px solid #5469d4;
            border-radius: 50%;
            width: 40px;
            height: 40px;
            animation: spin 1s linear infinite;
            margin: 0 auto 16px;
          }

          @keyframes spin {
            0% { transform: rotate(0deg); }
            100% { transform: rotate(360deg); }
          }

          .molam-footer {
            margin-top: 20px;
            text-align: center;
            font-size: 12px;
            color: #999;
          }

          .molam-footer a {
            color: #5469d4;
            text-decoration: none;
          }

          .molam-card-icons {
            display: flex;
            gap: 8px;
            margin-top: 8px;
          }

          .molam-card-icon {
            width: 40px;
            height: 25px;
            border: 1px solid #d0d0d0;
            border-radius: 3px;
            display: flex;
            align-items: center;
            justify-content: center;
            font-size: 10px;
            color: #666;
          }
        </style>
      `;

      this.shadowRoot.innerHTML = styles + `
        <div class="molam-container">
          <div id="content"></div>
        </div>
      `;
    }

    showLoading() {
      const content = this.shadowRoot.querySelector('#content');
      content.innerHTML = `
        <div class="molam-loading">
          <div class="molam-spinner"></div>
          <div>Initializing payment...</div>
        </div>
      `;
    }

    showPaymentForm() {
      const content = this.shadowRoot.querySelector('#content');
      const amount = this.paymentIntent.amount;
      const currency = this.paymentIntent.currency;

      content.innerHTML = `
        <div class="molam-header">
          <div class="molam-amount">${this.formatAmount(amount, currency)}</div>
          <div class="molam-description">${this.paymentIntent.description || ''}</div>
        </div>

        <div id="error-container"></div>

        <form id="payment-form">
          <div class="molam-form-group">
            <label class="molam-label">Card Number</label>
            <input type="text" id="card-number" class="molam-input" placeholder="4242 4242 4242 4242" maxlength="19" autocomplete="cc-number" required>
            <div class="molam-card-icons">
              <div class="molam-card-icon">VISA</div>
              <div class="molam-card-icon">MC</div>
              <div class="molam-card-icon">AMEX</div>
            </div>
          </div>

          <div class="molam-card-row">
            <div class="molam-form-group">
              <label class="molam-label">Expiry (MM/YY)</label>
              <input type="text" id="card-expiry" class="molam-input" placeholder="12/25" maxlength="5" autocomplete="cc-exp" required>
            </div>
            <div class="molam-form-group">
              <label class="molam-label">CVC</label>
              <input type="text" id="card-cvc" class="molam-input" placeholder="123" maxlength="4" autocomplete="cc-csc" required>
            </div>
          </div>

          <div class="molam-form-group">
            <label class="molam-label">Cardholder Name</label>
            <input type="text" id="card-name" class="molam-input" placeholder="John Doe" autocomplete="cc-name" required>
          </div>

          <button type="submit" class="molam-button" id="submit-button">
            Pay ${this.formatAmount(amount, currency)}
          </button>
        </form>

        <div class="molam-footer">
          Powered by <a href="https://molam.com" target="_blank">Molam</a>
        </div>
      `;

      this.attachFormHandlers();
    }

    attachFormHandlers() {
      const form = this.shadowRoot.querySelector('#payment-form');
      const cardNumberInput = this.shadowRoot.querySelector('#card-number');
      const expiryInput = this.shadowRoot.querySelector('#card-expiry');
      const cvcInput = this.shadowRoot.querySelector('#card-cvc');
      const submitButton = this.shadowRoot.querySelector('#submit-button');

      // Format card number
      cardNumberInput.addEventListener('input', (e) => {
        let value = e.target.value.replace(/\s/g, '');
        value = value.replace(/(\d{4})/g, '$1 ').trim();
        e.target.value = value;
      });

      // Format expiry
      expiryInput.addEventListener('input', (e) => {
        let value = e.target.value.replace(/\D/g, '');
        if (value.length >= 2) {
          value = value.substring(0, 2) + '/' + value.substring(2, 4);
        }
        e.target.value = value;
      });

      // Only allow digits for CVC
      cvcInput.addEventListener('input', (e) => {
        e.target.value = e.target.value.replace(/\D/g, '');
      });

      // Handle form submission
      form.addEventListener('submit', async (e) => {
        e.preventDefault();
        await this.handlePayment();
      });
    }

    async handlePayment() {
      const errorContainer = this.shadowRoot.querySelector('#error-container');
      const submitButton = this.shadowRoot.querySelector('#submit-button');
      errorContainer.innerHTML = '';

      const cardNumber = this.shadowRoot.querySelector('#card-number').value.replace(/\s/g, '');
      const cardExpiry = this.shadowRoot.querySelector('#card-expiry').value;
      const cardCvc = this.shadowRoot.querySelector('#card-cvc').value;
      const cardName = this.shadowRoot.querySelector('#card-name').value;

      // Basic validation
      if (!this.validateCardNumber(cardNumber)) {
        this.showError('Invalid card number');
        return;
      }

      const [expMonth, expYear] = cardExpiry.split('/');
      if (!expMonth || !expYear || expMonth < 1 || expMonth > 12) {
        this.showError('Invalid expiry date');
        return;
      }

      if (cardCvc.length < 3) {
        this.showError('Invalid CVC');
        return;
      }

      try {
        submitButton.disabled = true;
        submitButton.textContent = 'Processing...';

        // Tokenize payment method
        const paymentMethod = await this.sdk.tokenizePaymentMethod({
          number: cardNumber,
          expMonth: expMonth,
          expYear: '20' + expYear,
          cvc: cardCvc,
          name: cardName
        });

        // Confirm payment intent
        const result = await this.sdk.confirmPaymentIntent(
          this.paymentIntent.intent_reference,
          paymentMethod.token
        );

        this.showSuccess(result);
        this.dispatchEvent(new CustomEvent('payment-success', { detail: result }));

      } catch (error) {
        this.showError(error.message);
        this.dispatchEvent(new CustomEvent('payment-error', { detail: { error: error.message } }));
        submitButton.disabled = false;
        submitButton.textContent = `Pay ${this.formatAmount(this.paymentIntent.amount, this.paymentIntent.currency)}`;
      }
    }

    validateCardNumber(number) {
      // Luhn algorithm
      if (!/^\d{13,19}$/.test(number)) return false;

      let sum = 0;
      let isEven = false;

      for (let i = number.length - 1; i >= 0; i--) {
        let digit = parseInt(number[i], 10);

        if (isEven) {
          digit *= 2;
          if (digit > 9) digit -= 9;
        }

        sum += digit;
        isEven = !isEven;
      }

      return sum % 10 === 0;
    }

    showError(message) {
      const errorContainer = this.shadowRoot.querySelector('#error-container');
      errorContainer.innerHTML = `<div class="molam-error">${message}</div>`;
      this.sdk?.logEvent('error_displayed', { error: message });
    }

    showSuccess(result) {
      const content = this.shadowRoot.querySelector('#content');
      content.innerHTML = `
        <div class="molam-success">
          <div style="font-size: 48px; margin-bottom: 12px;">✓</div>
          <div style="font-size: 18px; font-weight: 600; margin-bottom: 8px;">Payment Successful!</div>
          <div style="font-size: 14px;">Transaction ID: ${result.intent_reference}</div>
        </div>
        <div class="molam-footer" style="margin-top: 16px;">
          Powered by <a href="https://molam.com" target="_blank">Molam</a>
        </div>
      `;
    }

    formatAmount(amount, currency) {
      return new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: currency
      }).format(amount);
    }
  }

  // Register custom element
  if (!customElements.get('molam-checkout')) {
    customElements.define('molam-checkout', MolamCheckoutElement);
  }

  // Expose SDK globally
  window.Molam = MolamSDK;

  // Log SDK initialization
  console.log(`Molam Form SDK v${SDK_VERSION} loaded`);

})();

File: brique-94/src/sdk/mobile/flutter/molam_form.dart
/// Molam Form Core - Flutter SDK
/// Universal payment widget for mobile integration (iOS & Android)
///
/// Usage:
/// ```dart
/// MolamCheckout(
///   publishableKey: 'pk_test_xxx',
///   amount: 99.99,
///   currency: 'USD',
///   customerEmail: 'customer@example.com',
///   onSuccess: (result) => print('Payment successful: $result'),
///   onError: (error) => print('Payment failed: $error'),
/// )
/// ```

library molam_form;

import 'dart:convert';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;

const String SDK_VERSION = '1.0.0';
const String API_BASE_URL = 'https://api.molam.com/form';

/// Molam SDK Client
class MolamSDK {
  final String publishableKey;
  final String environment;
  final http.Client _client = http.Client();

  MolamSDK(this.publishableKey)
      : environment = publishableKey.startsWith('pk_test_') ? 'test' : 'live' {
    if (!publishableKey.startsWith('pk_')) {
      throw ArgumentError('Invalid publishable key. Must start with "pk_"');
    }
  }

  /// Create a payment intent
  Future<Map<String, dynamic>> createPaymentIntent({
    required double amount,
    required String currency,
    String? customerEmail,
    String? customerName,
    String? description,
    Map<String, dynamic>? metadata,
  }) async {
    final response = await _client.post(
      Uri.parse('$API_BASE_URL/payment-intents'),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer $publishableKey',
      },
      body: jsonEncode({
        'amount': amount,
        'currency': currency.toUpperCase(),
        'customer_email': customerEmail,
        'customer_name': customerName,
        'description': description,
        'metadata': metadata ?? {},
      }),
    );

    if (response.statusCode != 201) {
      final error = jsonDecode(response.body);
      throw Exception(error['message'] ?? 'Failed to create payment intent');
    }

    final intent = jsonDecode(response.body);
    await _logEvent('intent_created', {'intent_reference': intent['intent_reference']});
    return intent;
  }

  /// Confirm a payment intent
  Future<Map<String, dynamic>> confirmPaymentIntent(
    String intentId,
    String paymentMethodToken,
  ) async {
    final response = await _client.patch(
      Uri.parse('$API_BASE_URL/payment-intents/$intentId'),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer $publishableKey',
      },
      body: jsonEncode({
        'action': 'confirm',
        'payment_method_token': paymentMethodToken,
      }),
    );

    if (response.statusCode != 200) {
      final error = jsonDecode(response.body);
      throw Exception(error['message'] ?? 'Failed to confirm payment intent');
    }

    final intent = jsonDecode(response.body);
    await _logEvent('intent_confirmed', {'intent_reference': intent['intent_reference']});
    return intent;
  }

  /// Tokenize payment method (mock - in production use platform channels)
  Future<Map<String, dynamic>> tokenizePaymentMethod({
    required String cardNumber,
    required String expMonth,
    required String expYear,
    required String cvc,
    required String cardholderName,
  }) async {
    // Mock tokenization - in production, use native SDKs or platform channels
    final token = 'pm_${environment}_${DateTime.now().millisecondsSinceEpoch}_${_generateRandomString(9)}';
    final last4 = cardNumber.substring(cardNumber.length - 4);
    final brand = _detectCardBrand(cardNumber);

    await _logEvent('payment_method_tokenized', {
      'token_prefix': token.substring(0, 10),
      'card_last4': last4,
      'brand': brand,
    });

    return {
      'token': token,
      'card': {
        'last4': last4,
        'brand': brand,
        'exp_month': expMonth,
        'exp_year': expYear,
      },
    };
  }

  /// Log telemetry event
  Future<void> _logEvent(String eventType, Map<String, dynamic> payload) async {
    try {
      await _client.post(
        Uri.parse('$API_BASE_URL/logs'),
        headers: {
          'Content-Type': 'application/json',
          'Authorization': 'Bearer $publishableKey',
        },
        body: jsonEncode({
          'event_type': eventType,
          'sdk_version': SDK_VERSION,
          'platform': 'flutter',
          'payload': payload,
        }),
      );
    } catch (e) {
      // Silent fail for logging
      debugPrint('Failed to log event: $e');
    }
  }

  /// Detect card brand from number
  String _detectCardBrand(String number) {
    if (number.startsWith('4')) return 'visa';
    if (RegExp(r'^5[1-5]').hasMatch(number)) return 'mastercard';
    if (RegExp(r'^3[47]').hasMatch(number)) return 'amex';
    if (RegExp(r'^6(?:011|5)').hasMatch(number)) return 'discover';
    return 'unknown';
  }

  String _generateRandomString(int length) {
    const chars = 'abcdefghijklmnopqrstuvwxyz0123456789';
    return List.generate(length, (index) => chars[DateTime.now().millisecond % chars.length]).join();
  }

  void dispose() {
    _client.close();
  }
}

/// Molam Checkout Widget
class MolamCheckout extends StatefulWidget {
  final String publishableKey;
  final double amount;
  final String currency;
  final String? customerEmail;
  final String? customerName;
  final String? description;
  final Function(Map<String, dynamic>) onSuccess;
  final Function(String) onError;
  final Color? primaryColor;

  const MolamCheckout({
    Key? key,
    required this.publishableKey,
    required this.amount,
    required this.currency,
    required this.onSuccess,
    required this.onError,
    this.customerEmail,
    this.customerName,
    this.description,
    this.primaryColor,
  }) : super(key: key);

  @override
  State<MolamCheckout> createState() => _MolamCheckoutState();
}

class _MolamCheckoutState extends State<MolamCheckout> {
  late MolamSDK _sdk;
  Map<String, dynamic>? _paymentIntent;
  bool _isLoading = true;
  bool _isProcessing = false;
  String? _errorMessage;
  bool _paymentSuccessful = false;

  final _formKey = GlobalKey<FormState>();
  final _cardNumberController = TextEditingController();
  final _expiryController = TextEditingController();
  final _cvcController = TextEditingController();
  final _nameController = TextEditingController();

  @override
  void initState() {
    super.initState();
    _sdk = MolamSDK(widget.publishableKey);
    _initializePayment();
  }

  @override
  void dispose() {
    _cardNumberController.dispose();
    _expiryController.dispose();
    _cvcController.dispose();
    _nameController.dispose();
    _sdk.dispose();
    super.dispose();
  }

  Future<void> _initializePayment() async {
    try {
      final intent = await _sdk.createPaymentIntent(
        amount: widget.amount,
        currency: widget.currency,
        customerEmail: widget.customerEmail,
        customerName: widget.customerName,
        description: widget.description ?? 'Payment of ${widget.amount} ${widget.currency}',
      );

      setState(() {
        _paymentIntent = intent;
        _isLoading = false;
      });
    } catch (e) {
      setState(() {
        _errorMessage = 'Failed to initialize payment: ${e.toString()}';
        _isLoading = false;
      });
      widget.onError(_errorMessage!);
    }
  }

  Future<void> _handlePayment() async {
    if (!_formKey.currentState!.validate()) return;

    setState(() {
      _isProcessing = true;
      _errorMessage = null;
    });

    try {
      // Tokenize payment method
      final cardNumber = _cardNumberController.text.replaceAll(' ', '');
      final expiry = _expiryController.text.split('/');
      final paymentMethod = await _sdk.tokenizePaymentMethod(
        cardNumber: cardNumber,
        expMonth: expiry[0].trim(),
        expYear: '20${expiry[1].trim()}',
        cvc: _cvcController.text,
        cardholderName: _nameController.text,
      );

      // Confirm payment intent
      final result = await _sdk.confirmPaymentIntent(
        _paymentIntent!['intent_reference'],
        paymentMethod['token'],
      );

      setState(() {
        _paymentSuccessful = true;
        _isProcessing = false;
      });

      widget.onSuccess(result);
    } catch (e) {
      setState(() {
        _errorMessage = e.toString().replaceAll('Exception: ', '');
        _isProcessing = false;
      });
      widget.onError(_errorMessage!);
    }
  }

  @override
  Widget build(BuildContext context) {
    final primaryColor = widget.primaryColor ?? const Color(0xFF5469D4);

    if (_isLoading) {
      return _buildLoadingState();
    }

    if (_paymentSuccessful) {
      return _buildSuccessState(primaryColor);
    }

    return Card(
      elevation: 4,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
      child: Padding(
        padding: const EdgeInsets.all(24.0),
        child: Form(
          key: _formKey,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            mainAxisSize: MainAxisSize.min,
            children: [
              // Header
              Text(
                _formatAmount(widget.amount, widget.currency),
                style: const TextStyle(
                  fontSize: 32,
                  fontWeight: FontWeight.bold,
                ),
                textAlign: TextAlign.center,
              ),
              if (widget.description != null) ...[
                const SizedBox(height: 8),
                Text(
                  widget.description!,
                  style: TextStyle(
                    fontSize: 14,
                    color: Colors.grey[600],
                  ),
                  textAlign: TextAlign.center,
                ),
              ],
              const SizedBox(height: 24),

              // Error message
              if (_errorMessage != null) ...[
                Container(
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(
                    color: Colors.red[50],
                    border: Border.all(color: Colors.red[200]!),
                    borderRadius: BorderRadius.circular(8),
                  ),
                  child: Text(
                    _errorMessage!,
                    style: const TextStyle(color: Colors.red),
                  ),
                ),
                const SizedBox(height: 16),
              ],

              // Card number
              TextFormField(
                controller: _cardNumberController,
                decoration: InputDecoration(
                  labelText: 'Card Number',
                  hintText: '4242 4242 4242 4242',
                  border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
                  prefixIcon: const Icon(Icons.credit_card),
                ),
                keyboardType: TextInputType.number,
                maxLength: 19,
                onChanged: (value) {
                  final formatted = _formatCardNumber(value);
                  if (formatted != value) {
                    _cardNumberController.value = TextEditingValue(
                      text: formatted,
                      selection: TextSelection.collapsed(offset: formatted.length),
                    );
                  }
                },
                validator: (value) {
                  if (value == null || value.isEmpty) {
                    return 'Card number is required';
                  }
                  final digits = value.replaceAll(' ', '');
                  if (digits.length < 13 || digits.length > 19) {
                    return 'Invalid card number';
                  }
                  if (!_validateLuhn(digits)) {
                    return 'Invalid card number';
                  }
                  return null;
                },
              ),
              const SizedBox(height: 16),

              // Expiry and CVC
              Row(
                children: [
                  Expanded(
                    child: TextFormField(
                      controller: _expiryController,
                      decoration: InputDecoration(
                        labelText: 'Expiry (MM/YY)',
                        hintText: '12/25',
                        border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
                      ),
                      keyboardType: TextInputType.number,
                      maxLength: 5,
                      onChanged: (value) {
                        if (value.length == 2 && !value.contains('/')) {
                          _expiryController.text = '$value/';
                          _expiryController.selection = TextSelection.fromPosition(
                            TextPosition(offset: _expiryController.text.length),
                          );
                        }
                      },
                      validator: (value) {
                        if (value == null || value.isEmpty) {
                          return 'Expiry is required';
                        }
                        if (!RegExp(r'^\d{2}/\d{2}$').hasMatch(value)) {
                          return 'Invalid format (MM/YY)';
                        }
                        return null;
                      },
                    ),
                  ),
                  const SizedBox(width: 16),
                  Expanded(
                    child: TextFormField(
                      controller: _cvcController,
                      decoration: InputDecoration(
                        labelText: 'CVC',
                        hintText: '123',
                        border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
                      ),
                      keyboardType: TextInputType.number,
                      maxLength: 4,
                      obscureText: true,
                      validator: (value) {
                        if (value == null || value.isEmpty) {
                          return 'CVC is required';
                        }
                        if (value.length < 3) {
                          return 'Invalid CVC';
                        }
                        return null;
                      },
                    ),
                  ),
                ],
              ),
              const SizedBox(height: 16),

              // Cardholder name
              TextFormField(
                controller: _nameController,
                decoration: InputDecoration(
                  labelText: 'Cardholder Name',
                  hintText: 'John Doe',
                  border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
                  prefixIcon: const Icon(Icons.person),
                ),
                textCapitalization: TextCapitalization.words,
                validator: (value) {
                  if (value == null || value.isEmpty) {
                    return 'Cardholder name is required';
                  }
                  return null;
                },
              ),
              const SizedBox(height: 24),

              // Submit button
              ElevatedButton(
                onPressed: _isProcessing ? null : _handlePayment,
                style: ElevatedButton.styleFrom(
                  backgroundColor: primaryColor,
                  foregroundColor: Colors.white,
                  padding: const EdgeInsets.symmetric(vertical: 16),
                  shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
                  elevation: 2,
                ),
                child: _isProcessing
                    ? const SizedBox(
                        height: 20,
                        width: 20,
                        child: CircularProgressIndicator(
                          strokeWidth: 2,
                          valueColor: AlwaysStoppedAnimation<Color>(Colors.white),
                        ),
                      )
                    : Text(
                        'Pay ${_formatAmount(widget.amount, widget.currency)}',
                        style: const TextStyle(fontSize: 16, fontWeight: FontWeight.w600),
                      ),
              ),
              const SizedBox(height: 16),

              // Footer
              Text(
                'Powered by Molam',
                style: TextStyle(fontSize: 12, color: Colors.grey[600]),
                textAlign: TextAlign.center,
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildLoadingState() {
    return const Card(
      elevation: 4,
      child: Padding(
        padding: EdgeInsets.all(48.0),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            CircularProgressIndicator(),
            SizedBox(height: 16),
            Text('Initializing payment...'),
          ],
        ),
      ),
    );
  }

  Widget _buildSuccessState(Color primaryColor) {
    return Card(
      elevation: 4,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
      child: Padding(
        padding: const EdgeInsets.all(48.0),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Icon(Icons.check_circle, color: primaryColor, size: 64),
            const SizedBox(height: 16),
            const Text(
              'Payment Successful!',
              style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
            ),
            const SizedBox(height: 8),
            Text(
              'Transaction ID: ${_paymentIntent!['intent_reference']}',
              style: TextStyle(fontSize: 14, color: Colors.grey[600]),
            ),
          ],
        ),
      ),
    );
  }

  String _formatAmount(double amount, String currency) {
    return '$currency ${amount.toStringAsFixed(2)}';
  }

  String _formatCardNumber(String value) {
    final digits = value.replaceAll(' ', '');
    final buffer = StringBuffer();
    for (int i = 0; i < digits.length; i++) {
      if (i > 0 && i % 4 == 0) buffer.write(' ');
      buffer.write(digits[i]);
    }
    return buffer.toString();
  }

  bool _validateLuhn(String number) {
    int sum = 0;
    bool isEven = false;

    for (int i = number.length - 1; i >= 0; i--) {
      int digit = int.parse(number[i]);

      if (isEven) {
        digit *= 2;
        if (digit > 9) digit -= 9;
      }

      sum += digit;
      isEven = !isEven;
    }

    return sum % 10 == 0;
  }
}
